pkg/geo/geogfn/topology_operations.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geogfn
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/cockroachdb/errors"
+	"github.com/golang/geo/r3"
+	"github.com/golang/geo/s2"
+	"github.com/twpayne/go-geom"
+)
+
+// Centroid returns the Centroid of a given Geography.
+//
+// Note: In the case of (Multi)Polygon Centroid result, don't mirror with
+// `PostGIS`'s result. Though we are using the same algorithm. Their
+// implementation differs, as in the `PostGIS` reference point for all the
+// triangles remains same, which changes the centroid significantly as we
+// change the starting point of the outermost ring.
+func Centroid(g *geo.Geography, useSphereOrSpheroid UseSphereOrSpheroid) (*geo.Geography, error) {
+	geomRepr, err := g.AsGeomT()
+	if err != nil {
+		return nil, err
+	}
+	if geomRepr.Empty() {
+		// Return Empty GeometryCollection in case of Empty Geographical Objects
+		return geo.NewGeographyFromGeom(geom.NewGeometryCollection().SetSRID(geomRepr.SRID()))
+	}
+	switch geomRepr.(type) {
+	case *geom.Point, *geom.LineString, *geom.Polygon, *geom.MultiPoint, *geom.MultiLineString, *geom.MultiPolygon:
+	default:
+		// GeometryCollection is among the unhandled Geography type.
+		return nil, errors.Newf("unhandled geography type %T", geomRepr)
+	}
+
+	regions, err := geo.S2RegionsFromGeom(geomRepr, geo.EmptyBehaviorOmit)
+	if err != nil {
+		return nil, err
+	}
+	spheroid, err := g.Spheroid()
+	if err != nil {
+		return nil, err
+	}
+
+	// `localWgtCentroids` is the collection of all the centroid corresponds to
+	// various small regions in which we divide the given region for calculation
+	// of centroid. The magnitude of each vector(`s2.Point.Vector`) represents
+	// the weight corresponding to its region.
+	var localWgtCentroids []s2.Point
```

Comment:
Is Wgt weight? Mind using the full name `localWeightedCentroids`?

Also nit that the `localWgtCentroids`  and `s2.Point.Vector` variable names doesn't need the backticks.

Reply:
Sure,
Just out of curiosity, Do we have any standard rule when to use backticks?

Reply:
not particularly, i think generally prefer backticks around READMEs but not so much in code.